src/grpc/mod.rs

```diff
@@ -0,0 +1,149 @@
+// Copyright (c) 2021 Alibaba Cloud
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+use futures::channel::oneshot;
+use futures::executor::block_on;
+use futures::prelude::*;
+use grpcio::{Environment, ServerBuilder};
+use std::error::Error;
+use std::io::Read;
+use std::str;
+use std::sync::Arc;
+use std::{io, thread};
+use string_error::into_err;
+
+pub mod grpc_keyprovider;
+
+use grpc_keyprovider::keyprovider;
+use grpc_keyprovider::keyprovider_grpc::*;
+use grpc_keyprovider::keyprovider_structs::*;
+
+use super::kbc_manager;
+
+#[derive(Clone)]
+pub struct KeyProvider;
+
+impl KeyProviderService for KeyProvider {
+    fn un_wrap_key(
+        &mut self,
+        ctx: ::grpcio::RpcContext,
+        req: keyprovider::keyProviderKeyWrapProtocolInput,
+        sink: ::grpcio::UnarySink<keyprovider::keyProviderKeyWrapProtocolOutput>,
+    ) {
+        // Deserialize gRPC input to get kbs_annotation_packet.
+        let (annotation, dc) =
+            match get_annotation_and_dc_from_input(req.KeyProviderKeyWrapProtocolInput) {
+                Ok((a, d)) => (a, d),
+                Err(e) => {
+                    println!(
+                        "Some error happend when get annotation and dc from input: {}",
+                        e
+                    );
+                    return ();
```

Comment:
Is there another way to do this error handling? Do we want to propagate the error back to ocicrypt? Is there a more succinct way than a match? Maybe we should at least have the message point out that the error is coming from the Attestation Agent.

Reply:
At present, the communication protocol between AA and ocicrypt does not have a field for transmitting error information. We have put forward this suggestion to the development team of ocicrypt. They said that the communication protocol between AA and ocicrypt will be modified in the future to increase the field for transmitting error information. The ocicrypt of both the golang version and the rust version will be modified accordingly. AA will improve our error handling system after the upstream protocol is modified.